skyportal/models.py

```diff
@@ -3120,6 +3126,327 @@ class SourceNotification(Base):
+
+    dateobs = sa.Column(
+        sa.DateTime,
+        sa.ForeignKey(GcnEvent.dateobs),
+        nullable=False,
+        comment='UTC event timestamp',
+    )
+
+    content = deferred(
+        sa.Column(sa.LargeBinary, nullable=False, comment='Raw VOEvent content')
+    )
+
+    def _get_property(self, property_name, value=None):
+        root = lxml.etree.fromstring(self.content)
+        path = ".//Param[@name='{}']".format(property_name)
+        elem = root.find(path)
+        value = float(elem.attrib.get('value', '')) * 100
+        return value
+
+    @property
+    def has_ns(self):
+        return self._get_property(property_name="HasNS")
+
+    @property
+    def has_remnant(self):
+        return self._get_property(property_name="HasRemnant")
+
+    @property
+    def far(self):
+        return self._get_property(property_name="FAR")
+
+    @property
+    def bns(self):
+        return self._get_property(property_name="BNS")
+
+    @property
+    def nsbh(self):
+        return self._get_property(property_name="NSBH")
+
+    @property
+    def bbh(self):
+        return self._get_property(property_name="BBH")
+
+    @property
+    def mass_gap(self):
+        return self._get_property(property_name="MassGap")
+
+    @property
+    def noise(self):
+        return self._get_property(property_name="Terrestrial")
+
+
+class Localization(Base):
+    """Localization information, including the localization ID, event ID, right
+    ascension, declination, error radius (if applicable), and the healpix
+    map."""
+
+    nside = 512
+    """HEALPix resolution used for flat (non-multiresolution) operations."""
+
+    id = sa.Column(sa.Integer, primary_key=True, autoincrement=True)
+
+    dateobs = sa.Column(
+        sa.DateTime,
+        sa.ForeignKey(GcnEvent.dateobs),
+        nullable=False,
+        comment='UTC event timestamp',
+    )
+
+    localization_name = sa.Column(
+        sa.String, primary_key=True, comment='Localization name'
+    )
+
+    uniq = deferred(
+        sa.Column(
+            sa.ARRAY(sa.BigInteger),
+            nullable=False,
+            comment='Multiresolution HEALPix UNIQ pixel index array',
+        )
+    )
+
+    probdensity = deferred(
+        sa.Column(
+            sa.ARRAY(sa.Float),
+            nullable=False,
+            comment='Multiresolution HEALPix probability density array',
+        )
+    )
+
+    distmu = deferred(
+        sa.Column(
+            sa.ARRAY(sa.Float), comment='Multiresolution HEALPix distance mu array'
+        )
+    )
+
+    distsigma = deferred(
+        sa.Column(
+            sa.ARRAY(sa.Float), comment='Multiresolution HEALPix distance sigma array'
+        )
+    )
+
+    distnorm = deferred(
+        sa.Column(
+            sa.ARRAY(sa.Float),
+            comment='Multiresolution HEALPix distance normalization array',
+        )
+    )
+
+    contour = deferred(sa.Column(sa.JSON, comment='GeoJSON contours'))
```

Comment:
Did you have a specific reasoning for using regular JSON here instead of JSONB?

Reply:
Nope converting to JSONB seems good.